OracleWebCenterPortal/dockerfiles/12.2.1.4/README.md

```diff
@@ -0,0 +1,308 @@
+    "live-restore": true,
+    "userland-proxy": false,
+    "no-new-privileges": true
+}
+```
+Restart docker daemon 
+```
+$ sudo systemctl daemon-reload
+$ sudo systemctl restart docker
+```
+
+##  Preparing to Run Oracle WebCenter Portal Docker Container
+Configure an environment before running the Oracle WebCenter Portal Docker container. You need to set up a communication network between containers on the same host and a WebCenter Content Server instance.
+
+##### A. Creating a user-defined network
+##### B. WebCenter Content Server (Optional)
+
+### A. Creating a user-defined network
+Create a user-defined network to enable containers to communicate by running the following command:
+
+```
+$ docker network create -d bridge WCPortalNET
+```
+
+### B. WebCenter Content Server (Optional)
+You need to have a running WebCenter Content Server on any machine. The Content server connection details are required for configuring WebCenter Portal during configuration process.
+
+**Note :** This step is required only when the WebCenter Portal uses Content Server as content repository.
+
+##  Running Oracle WebCenter Portal Docker Container
+To run the Oracle WebCenter Portal Docker container, you need to create:
+* Container to manage the Admin Server.
+* Container to manage the Managed Server.
+
+### 1. Creating containers for WebCenter Portal Server
+
+#### 1.1. Update the environment file 
+
+Create an environment file `webcenter.env.list` file, to define the parameters.
+
+Update the parameters inside `webcenter.env.list` as per your local setup.
+
+```
+#Database Configuration
+DB_DROP_AND_CREATE=<true or false>
+DB_CONNECTION_STRING=<Hostname/ContainerName>:<Database Port>/<DB_PDB>.<DB_DOMAIN>
+DB_RCUPREFIX=<RCU Prefix>
+DB_PASSWORD=<Database Password>
+DB_SCHEMA_PASSWORD=<Schema Password>
+
+#configure container
+ADMIN_SERVER_CONTAINER_NAME=<Admin Server Container Name>
+ADMIN_PORT=<Admin Server Port>
+ADMIN_PASSWORD=<Admin Server Password>
+ADMIN_USERNAME=<Admin Server User Name>
+MANAGED_SERVER_PORT=<Managed Server Port>
+
+# Configure Content Server
+CONFIGURE_UCM_CONNECTION=<true or false>
+#Valid option for socket type are socket,jaxws 
+UCM_SOCKET_TYPE=<UCM Socket Type >
+#Required if UCM_SOCKET_TYPE is jaxws
+UCM_URL=<Configure UCM URL If Socket type is jaxws>
+UCM_HOST=<UCM Host>
+UCM_PORT=<UCM Port>
+UCM_ADMIN_USER=<UCM Admin User>
+
+# Configure Elasticsearch Server
+SEARCH_APP_USERNAME=<Search User Name>
+SEARCH_APP_USER_PASSWORD=<Search User Password>
+```
+
+### 1.2. Admin Container (WCPAdminContainer)
+#### A. Creating and Running Admin Container
+
+Run the following command to create the Admin Server container:
+
+```
+$ docker run -i -t --name $ADMIN_SERVER_CONTAINER_NAME --network=WCPortalNET -p <Any Free Port>:$ADMIN_PORT -v $DATA_VOLUME:/u01/oracle/user_projects --env-file <directory>/webcenter.env.list $WCPortalImageName /bin/bash -c "/u01/oracle/container-scripts/configureOrStartAdminServer.sh; /bin/bash"
```

Comment:
Why are you running `/bin/bash` at the end? Why doesn't your container just continue to run the process it's meant to run? You should never have an open shell running because it it's not an init and can't grab signals properly.

Reply:
Further, why are you specifying a command on the `run` line anyway? Why doesn't your container image have the correct `ENTRYPOINT` or `CMD` so that this isn't necessary? You can use one of your many environment variables to decide which script to run automatically.